snack-shop-fe/src/App.tsx

```diff
@@ -1,35 +1,54 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
-import './App.css'
+import { useEffect, useState } from 'react';
 
-function App() {
-  const [count, setCount] = useState(0)
+const App = () => {
+  const [data, setData] = useState(null);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    const fetchDataForSnacks = async () => {
```

Comment:
I'm wondering whether it's worth adding tests already at this initial stage when we don't have much insight yet. 🤔 On the other hand, things will get quickly out of hand if we already start neglect now. Also, we could also have tests drive the implementation and hence only build the very next step we actually need. Feel free to make a judgment call for this change.

Reply:
Tests will come soon, but in following PRs.